### DISCUSSION TITLE
龙猫对于 PCL2 社区中 BUG 的定义是怎样的？

### DISCUSSION THREAD

**Discussion by Deep-Dark-Forest** [Meloong-Git/PCL#3812] [提问] [answered]
最近出现了很多标签为 bug 的 issue，很多人对它们是不是 bug 提出了质疑，很多人都说这些 bug 都是手动触发，不能算 bug。但是我想了想又觉得不对劲，比如，如果 PCL 中打开 GitHub 反馈页的那个按钮，点击以后给你跳转到了原神官网，那么这算不算 bug 呢？这是不是手动触发呢？
一个 issue 是否能算得上 bug，对于反馈者而言非常重要，因为这直接影响到他能否获取活跃橙。
因此，我们需要@LTCatt 对 bug 做出定义，以及一个 bug 是否为手动触发是否会影响这个反馈是否为 bug，至少在 PCL2 社区中生效。

**Comment by LTCatt** [comment, marked as answer]
**不要滥用手动触发这一说法。**

这个定义的 **唯一目的** 是阻止 “为了获取活跃橙，而刻意地去构造一些不正常的情况” 这一滥用行为的。这是一个为阻止滥用而存在的概念，绝对不是用来筛选非恶意的普通反馈的。
明显的 “构造 bug” 的情况例如：上连点器、执行命令（批处理、发送错误 Windows 命令）、刻意的异常文件环境（加只读、上网络驱动器、加文件链接）、用错误的东西 “测试” PCL（使用损坏的压缩包、不遵守规范的 json 或 mod）等。

具体例子：
“对 PCL 窗口发送 Windows 窗口关闭命令会导致 PCL 崩溃” 属于手动触发
“在游戏实际启动的 10ms 内点击取消启动，会让 PCL 提示已取消，但游戏也启动了” 不属于手动触发
“把 Java 放在 U 盘里，游戏启动后拔出 U 盘，会导致游戏崩溃” 属于手动触发，因为触发条件是明显异常且人为的
“在下载结束的同时点击下载管理按钮会导致报错” 不属于手动触发
“使用 1ms 的连点器点击开始下载按钮会导致同时下载两个版本” 属于手动触发，同样因为触发条件是明显异常且人为的

关于普通反馈是否是 bug，取决于 **在代码上这一行为和我编写时的预期行为是否相符** ，你也可以认为这是我的主观判断（

**Reply by iPodtouch6557** [reply]
就是说正常人有概率干出来的就算对吧（把java放在U盘里肯定不是，谁会插着个大棍子玩电脑啊）

**Reply by wuliaodexiaoluo** [reply]
并且还是启动慢的一批
